test/repl.jl

```diff
@@ -177,29 +203,6 @@ begin
     hp = REPL.REPLHistoryProvider(Dict{Symbol,Any}(:julia => repl_mode,
                                                    :shell => shell_mode,
                                                    :help  => help_mode))
-    fakehistory =
-    """
-    # time: 2014-06-30 17:32:49 EDT
-    # mode: julia
-    \tshell
-    # time: 2014-06-30 17:32:59 EDT
-    # mode: shell
-    \tll
-    # time: 2014-06-30 17:32:49 EDT
-    # mode: julia
-    \t1 + 1
-    # time: 2014-06-30 17:35:39 EDT
-    # mode: foobar
-    \tbarfoo
-    # time: 2014-06-30 18:44:29 EDT
-    # mode: shell
-    \tls
-    # time: 2014-06-30 19:44:29 EDT
-    # mode: foobar
-    \tls
-    # time: 2014-06-30 20:44:29 EDT
-    # mode: julia
-    \t2 + 2"""
```

Comment:
This all comes from @nolta's change, https://github.com/JuliaLang/julia/commit/f8b3f9e0ad730bed588227f476993ad021241943
If the last line is not empty after stripping, it walks through every non-blank line, and figures out the minimum indentation and uses that for unindent.
It looks like a lot of code depends on that behavior also.